compiler/main.nim

```diff
@@ -119,6 +119,10 @@ proc commandCompileToJS(graph: ModuleGraph) =
     if conf.outFile.isEmpty:
       conf.outFile = RelativeFile(conf.projectName & ".js")
 
+    # js backend will ignore threads options
```

Comment:
i think the correct fix is similar to the one done for gc:arc:
```nim
  of "gc":
    if conf.backend == backendJs: return # for: bug #16033
```

so in this case it'd be:
```nim
  of "threadanalysis":
    if conf.backend == backendJs: discard
    else: processOnOffSwitchG(conf, {optThreadAnalysis}, arg, pass, info)
...
  of "threads":
    if conf.backend == backendJs: discard
    else: processOnOffSwitchG(conf, {optThreads}, arg, pass, info)
```

that way there's no intermediate bad state.

(for more background see https://github.com/nim-lang/Nim/pull/16036 and its followup https://github.com/nim-lang/Nim/pull/16056)

Reply:
make sense

Reply:
thanks